storage/innobase/srv/srv0start.cc

```diff
@@ -688,6 +688,9 @@ static dberr_t srv_undo_tablespace_read_encryption(pfs_os_file_t fh,
 
   /* Return if the encryption metadata is empty. */
   if (memcmp(first_page + offset, ENCRYPTION_KEY_MAGIC_V3,
+             ENCRYPTION_MAGIC_SIZE) != 0 &&
+      /* PS 5.7 undo encryption */
+      memcmp(first_page + offset, ENCRYPTION_KEY_MAGIC_V2,
```

Comment:
Shouldn't the exemption be given only when srv_is_upgrade_mode is true?

Reply:
DD where the upgrade mode is detected is initialized after the undo tablespaces are opened thus - no.

Reply:
This answer should be "yes". I am not convinced with the answer and I tried myself.  We can rely on the variable "srv_is_upgrade_mode".

(gdb) p srv_is_upgrade_mode
$4 = true

(gdb) p space->name
$5 = 0x7fffe2c63618 "innodb_undo001"

(gdb) p space->files
$6 = std::vector of length 1, capacity 1 = {{space = 0x7fffe2c63358, name = 0x7fffe2c63678 "./undo001", is_open = false, handle = {m_psi = 0x0,
      m_file = 0}, sync_event = 0x7fffe2c636a8, is_raw_disk = false, size = 640, flush_size = 640, init_size = 640, max_size = 4294967294,
    n_pending = 0, n_pending_flushes = 0, in_use = 0, modification_counter = 0, flush_counter = 0, LRU = {prev = 0x0, next = 0x0}, punch_hole = true,
    block_size = 4096, atomic_write = false, magic_n = 89389}}

#0  srv_undo_tablespace_read_encryption (fh=..., space=0x7fffe2c63358) at /home/satya/WORK/PS-8.0-5508/storage/innobase/srv/srv0start.cc:659
#1  0x0000555559c3b6c5 in srv_undo_tablespace_open (undo_space=...) at /home/satya/WORK/PS-8.0-5508/storage/innobase/srv/srv0start.cc:949
#2  0x0000555559c3ba14 in srv_undo_tablespace_open_by_id (space_id=1) at /home/satya/WORK/PS-8.0-5508/storage/innobase/srv/srv0start.cc:997
#3  0x0000555559c3c03b in srv_undo_tablespaces_open () at /home/satya/WORK/PS-8.0-5508/storage/innobase/srv/srv0start.cc:1090
#4  0x0000555559c3d4d1 in srv_undo_tablespaces_init (create_new_db=false) at /home/satya/WORK/PS-8.0-5508/storage/innobase/srv/srv0start.cc:1506
#5  0x0000555559c40ef2 in srv_start (create_new_db=false, scan_directories="./;./;./")
    at /home/satya/WORK/PS-8.0-5508/storage/innobase/srv/srv0start.cc:2755
#6  0x0000555559988a5c in innobase_init_files (dict_init_mode=DICT_INIT_UPGRADE_57_FILES, tablespaces=0x7fffe865f930)
    at /home/satya/WORK/PS-8.0-5508/storage/innobase/handler/ha_innodb.cc:5356
#7  0x000055555999a0f5 in innobase_ddse_dict_init (dict_init_mode=DICT_INIT_UPGRADE_57_FILES, version=80014, tables=0x7fffe865f910, 
    tablespaces=0x7fffe865f930) at /home/satya/WORK/PS-8.0-5508/storage/innobase/handler/ha_innodb.cc:12586
#8  0x00005555594ab2e8 in dd::bootstrap::DDSE_dict_init (thd=0x55555c473f50, dict_init_mode=DICT_INIT_UPGRADE_57_FILES, version=80014)
    at /home/satya/WORK/PS-8.0-5508/sql/dd/impl/bootstrapper.cc:2320
#9  0x00005555596da55d in dd::upgrade_57::do_pre_checks_and_initialize_dd (thd=0x55555c473f50)
    at /home/satya/WORK/PS-8.0-5508/sql/dd/upgrade/upgrade.cc:1086
#10 0x000055555874dff2 in bootstrap::handle_bootstrap (arg=0x55555c473f50) at /home/satya/WORK/PS-8.0-5508/sql/bootstrap.cc:347
#11 0x0000555559892320 in pfs_spawn_thread (arg=0x55555c371320) at /home/satya/WORK/PS-8.0-5508/storage/perfschema/pfs.cc:2836
#12 0x00007ffff79b66db in start_thread (arg=0x7fffe8660700) at pthread_create.c:463
#13 0x00007ffff5e9388f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95



ls data/
auto.cnf      ib_buffer_pool  ib_logfile0  mtr    mysql_dd_upgrade_info  sys   undo001  undo003
binlog.index  ibdata1         ib_logfile1  mysql  performance_schema     test  undo002  xb_doublewrite

Reply:
sorry, I have messed up the condition when I was testing this. You are right.

Reply:
Thanks, Will wait for the updated patch